packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol

```diff
@@ -244,6 +276,25 @@ contract L2OutputOracle is Initializable, Semver {
         return l2Outputs[_l2OutputIndex];
     }
 
+    /**
+     * @notice Returns an output by block number.
+     *
+     * @param _l2BlockNumber L2 Block Number of the output to return.
+     *
+     * @return The output for the given block number.
+     */
+    function getL2OutputByNumber(uint256 _l2BlockNumber)
+        external
+        view
+        returns (Types.OutputProposal memory)
+    {
+        uint256 index = l2OutputIndices[_l2BlockNumber];
+        if (index == 0) {
+            return Types.OutputProposal({ outputRoot: bytes32(0), timestamp: 0, l2BlockNumber: 0 });
```

Comment:
Should this revert to indicate non existence rather than returning the null value?